### PR TITLE
Component Base / Modifier Improvements

### PR DESCRIPTION
**Pull request by macandcheese**
## Add base / modifier paradigm for Components, and update styles

This update changes how we display components in the documentation. Currently, everything is a modifier. This adds some attribute syntax in the table of contents to separate a “base” component and class from modifier classes. This allows us to separate the two in documentation, and place markup closer to the base component. We also display the base class for a component, if present.

### Big Stuff
- Add base / modifier paradigm to components
 - A "modifiers" attribute still must be present in order to display a component in docs. However, if you only have a base class, you can now set a more logical 'modifiers: false' instead of 'modifiers: true'. If you have modifiers, continue to list them as is the case now.

### Medium Stuff
- Adjust display of components on the docs page
- Set max-height for code blocks to clip super-long texts

### Small Stuff
- Add links and clickable section titles to components and docs for easier sharing

### CODE REVIEW

File: docs/source/assets/css/all.scss [hunk 1]
```diff
@@ -76,10 +76,14 @@ $include-calcite-icons: false;
 .styleguide-modifiers + pre {
   margin-top: 0;
   code {
+    border-top-width:0px;
```
Comment (paulcpederson) [inline docs/source/assets/css/all.scss]: space here

File: docs/source/assets/css/all.scss [hunk 2]
```diff
@@ -76,10 +76,14 @@ $include-calcite-icons: false;
...
     border-radius: 0 0 3px 3px;
+    max-height:300px;
   }
 }
 
+
```
Comment (paulcpederson) [inline docs/source/assets/css/all.scss]: probably don't need these

File: docs/source/assets/css/all.scss [hunk 3]
```diff
@@ -76,10 +76,14 @@ $include-calcite-icons: false;
 .styleguide-modifiers + pre {
   margin-top: 0;
   code {
+    border-top-width:0px;
     border-radius: 0 0 3px 3px;
+    max-height:300px;
```
Comment (paulcpederson) [inline docs/source/assets/css/all.scss]: also here

**Comment by paulcpederson** [comment]
@macandcheese couple tiny formatting things, but looks good!

File: docs/source/assets/css/all.scss (hunk 2 above)
Comment (macandcheese) [reply docs/source/assets/css/all.scss]: Good catch, fixed.

**Comment by paulcpederson** [comment]
✨

**Pull request merged**